blockstore.go

```diff
@@ -114,8 +114,22 @@ func NewBlockstore(d ds.Batching) Blockstore {
 	dd := dsns.Wrap(d, BlockPrefix)
 	dsb = dd
 	return &blockstore{
-		datastore: dsb,
-		rehash:    uatomic.NewBool(false),
+		datastore:  dsb,
+		rehash:     uatomic.NewBool(false),
+		checkFirst: true,
+	}
+}
+
+// NewWriteThrough returns a default Blockstore implementation
+// which does not tries to check if blocks exist prior to writing.
+func NewWriteThrough(d ds.Batching) Blockstore {
```

Comment:
It's very weird to define a function by something it does not do.

I would either add the functional option pattern to `NewBlockstore` or follow `HashOnRead` and add a `UseBloomFilter` or something like this.

Reply:
It's the same pattern as in BlockService ?

Reply:
I can add it as an option sure

Reply:
But this blockstore has no Options. There is also BlockstoreNoPrefix()... I shouldn't have to re-factor libraries that I don't know when adding simple features.

Reply:
Can you tell me exactly what I have to do so that my PR will be merged?

Reply:
We can merge this as-is, this isn't critical.
What I would do this is this:
```go
type Option struct{
  f func(*blockstore)
}

// CheckFirst will disable or enable weather or not we should check in the bloom hash first.
func CheckFirst(v bool) Option {
  return Option{func(b *blockstore) {
    b.checkFirst = false
  }}
}

// NewBlockstore returns a default Blockstore implementation
// using the provided datastore.Batching backend.
func NewBlockstore(d ds.Batching, opts ...Option) Blockstore {
	var dsb ds.Batching
	dd := dsns.Wrap(d, BlockPrefix)
	dsb = dd
	b := &blockstore{
		datastore:  dsb,
		rehash:       uatomic.NewBool(false),
		checkFirst: true,
	}
	
	for _, o := range opts {
	  o.f(b)
	}
	
	return b
}
```
we can also refactor other options like rehash or BlockstoreNoPrefix like this but I'm not asking you to refactor other options.

Adding more and more one off functions doesn't scale (what if I want a blockstore without prefix and without check first how do I do ?)

Reply:
ok, see now